### PR TITLE
feat: optimizing CustomLinuxConfig, KubeletConfigs and others

### PR DESCRIPTION
**Pull request by Devinwong**
Fixes # 

**Description**
Optimizing CustomLinuxConfig, KubeletConfigs and others. Tried to reduce the number of variables from NB contract.

**How was this change tested?**
Unit tests.
Manual scale-up tests.
*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/providers/imagefamily/bootstrap/parser_helpers.go [hunk 1]
```diff
@@ -245,3 +258,260 @@ func getHasSearchDomain(csd *nbcontractv1.CustomSearchDomain) bool {
...
+		m["net.ipv4.neigh.default.gc_thresh2"] = s.GetNetIpv4NeighDefaultGcThresh2()
+	}
+
+	if s.NetIpv4NeighDefaultGcThresh3 == nil {
+		m["net.ipv4.neigh.default.gc_thresh3"] = 16384
+	} else {
+		m["net.ipv4.neigh.default.gc_thresh3"] = s.GetNetIpv4NeighDefaultGcThresh3()
+	}
+
+	if s.NetCoreNetdevMaxBacklog != nil {
```
Comment (lilypan26) [inline pkg/providers/imagefamily/bootstrap/parser_helpers.go:345]: is there a reason why you can't directly use the getter's here and all the ones below? the getters are nil safe so i dont think you need the nil check.

Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/parser_helpers.go:345]: Yes there is a reason for this. What I wanted to do is to tell if the value is set as 0 intentionally by user (bootstrapper) or it's not set.
For example, if I use the getter here, either user sets it as 0 or user doesn't set it at all, the getter will still give me the default value 0. And then the logic will add an entry _"net.ipv4.neigh.default.gc_thresh3" = 0_ to the map but in fact it should be _"net.ipv4.neigh.default.gc_thresh3 = 16384"_ .

Comment (Devinwong) [reply pkg/providers/imagefamily/bootstrap/parser_helpers.go:345]: I left a comment in line 311 too in case later we want to know why.

Comment (lilypan26) [reply pkg/providers/imagefamily/bootstrap/parser_helpers.go:345]: ahh oops i missed that, okay makes sense!

**Review by lilypan26 — APPROVED** [review]

**Pull request merged**